lib/logstorage/datadb.go

```diff
@@ -1165,14 +1175,36 @@ func mustWritePartNames(path string, smallPartNames, bigPartNames []string) {
 
 func mustReadPartNames(path string) []string {
 	partNamesPath := filepath.Join(path, partsFilename)
-	data, err := os.ReadFile(partNamesPath)
-	if err != nil {
-		logger.Panicf("FATAL: cannot read %s: %s", partNamesPath, err)
+	if fs.IsPathExist(partNamesPath) {
```

Comment:
VictoriaMetrics checks if `parts.json` exists for different purpose. It could be the case for migration from version without this file.

 I don't think it's a good idea to have it for VictoriaLogs. If `parts.json` is missing, probably there was some data corruption and attempt to recover this file most probably will fail.

 Because parts may contain incomplete or corrupted data. For instance, VictoriaLogs performed background merge into some part and didn't finish it. Such part cannot be used and will trigger either openPart error or runtime crash during requests.

Reply:
> Though I don't see any case that could cause parts.json to disappear except during part initialization. The background merge can't do that, right?

 Background merge only updates it content. And I see no valid cases when `parts.json` can be removed.

 For me, most probable cases are - file was removed manually by mistake or file was removed due to disk fs corruption ( some extremely rare case). In my understanding, both cases require manual user action and automatic recover is not what expected.

Reply:
> The solution is to drop the partition directory when opening it if there is no datadb/parts.json file inside it. The code must ensure that the datadb directory doesn't contain any other subdirectories before dropping the partition directory. This will protect from deleting non-empty partition in case of other unexpected states, which might need additional attention from user's side or from our side.

 I think, it's a good way to go.